robotiq_controllers/src/robotiq_activation_controller.cpp

```diff
@@ -105,14 +105,19 @@ rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn Roboti
 bool RobotiqActivationController::reactivateGripper(std_srvs::srv::Trigger::Request::SharedPtr /*req*/,
                                                     std_srvs::srv::Trigger::Response::SharedPtr resp)
 {
-  command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].set_value(ASYNC_WAITING);
-  command_interfaces_[REACTIVATE_GRIPPER_CMD].set_value(1.0);
+  resp->success = command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].set_value(ASYNC_WAITING);
+  resp->success &= command_interfaces_[REACTIVATE_GRIPPER_CMD].set_value(1.0);
 
-  while (command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].get_value() == ASYNC_WAITING)
+  while (true)
   {
+    const auto maybe_value = command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].get_optional();
+    if (maybe_value && maybe_value.value() != ASYNC_WAITING)
+    {
+      break;
+    }
     std::this_thread::sleep_for(std::chrono::milliseconds(50));
   }
-  resp->success = command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].get_value();
+  resp->success &= command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].get_optional().has_value();
```

Comment:
I'm not familiar with this code here, but should this really be has_value()?
```suggestion
  if (resp->success &= command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].get_optional().has_value()) {
    resp->success &= command_interfaces_[REACTIVATE_GRIPPER_RESPONSE].get_optional().value();
  }
```
This would have more similar logic to before.

Reply:
So `get_value()` returns a double, I think this was being implicitly cast before which is why I changed it.

Reply:
In fact, with some of the other changes in this PR using the value instead of a boolean expression causes a compilation error!

Reply:
I see the problem with the cast, but get_optional.has_value() is not the same as get_value() before. (as I said, I don't know what info is actually in the interface).

Reply:
fair point, I kept the old behavior and left a comment for someone who knows what's supposed to happen to maybe fix it later